nearcore/src/lib.rs

```diff
@@ -74,8 +74,78 @@ pub fn get_default_home() -> PathBuf {
     PathBuf::default()
 }
 
+mod r#impl {
+    const MIGRATION_IS_IN_PROGRESS_STORE_KEY: &'static [u8] = b"migration_is_in_progress";
+    const STORE_IS_PROBABLY_CORRUPTED_STORE_KEY: &'static [u8] = b"store_is_probably_corrupted";
+    use near_store::Store;
+    use near_store::db::DBCol::ColBlockMisc;
+    use tracing::error;
+
+    pub struct MigrationIsInProgress<'a> {
+        store: &'a Store
+    }
+
+    impl<'a> MigrationIsInProgress<'a> {
+        pub fn new(store: &'a Store) -> MigrationIsInProgress {
+            let mut store_update = store.store_update();
+            store_update.set_ser(ColBlockMisc, &MIGRATION_IS_IN_PROGRESS_STORE_KEY, &true).unwrap();
+            store_update.commit().unwrap();
+            MigrationIsInProgress{store}
+        }
+    }
+
+    impl<'a> Drop for MigrationIsInProgress<'a> {
+        fn drop(&mut self) {
+            if !std::thread::panicking() {
+                let mut store_update = self.store.store_update();
+                store_update.set_ser(ColBlockMisc, &MIGRATION_IS_IN_PROGRESS_STORE_KEY, &false).unwrap();
+                store_update.commit().unwrap();
+            }
+        }
+    }
+
+    pub fn check_if_migration_is_in_progress(store: &Store, fail_if_migration_is_in_progress: bool) {
+        let store_is_probably_corrupted =
+            match store.get_ser::<bool>(ColBlockMisc, &STORE_IS_PROBABLY_CORRUPTED_STORE_KEY) {
+                Ok(Some(x)) => x,
+                Ok(None) => false,
+                Err(e) => panic!("Can't read DB, {:?}", e),
+            };
+        if store_is_probably_corrupted {
+            error!(concat!("The state was earlier migrated while another migration was interrupted",
+                           " so the current state can be corrupted and lead to undefined failures."));
+        }
+
+        let migration_is_in_progress =
+            match store.get_ser::<bool>(ColBlockMisc, &MIGRATION_IS_IN_PROGRESS_STORE_KEY) {
+                Ok(Some(x)) => x,
+                Ok(None) => false,
+                Err(e) => panic!("Can't read DB, {:?}", e),
+            };
+
+        if migration_is_in_progress {
+            error!(concat!("The state migration was earlier interrupted so the current state can ",
+                   "be corrupted, it's recommended to recover the node from a backup otherwise ",
+                   "this can lead to undefined failures."));
+            if fail_if_migration_is_in_progress {
+                panic!(concat!("Failed to start because the state is probably corrupted and ",
+                       "the flag fail_if_migration_is_in_progress is enabled"));
```

Comment:
I'd rephrase it - to say something like:

if you want to proceed anyway, rerun with --fail_if_migration_is_in_progress=false